hpy/devel/include/hpy/hpymodule.h

```diff
@@ -34,14 +34,18 @@ typedef struct {
 #ifdef HPY_UNIVERSAL_ABI
 
 // module initialization in the universal case
-#define HPy_MODINIT(modname)                                   \
-    _HPy_HIDDEN HPyContext *_ctx_for_trampolines;              \
-    static HPy init_##modname##_impl(HPyContext *ctx);         \
-    HPyMODINIT_FUNC                                         \
-    HPyInit_##modname(HPyContext *ctx)                     \
-    {                                                          \
-        _ctx_for_trampolines = ctx;                            \
-        return init_##modname##_impl(ctx);                     \
+#define HPy_MODINIT(modname)                                      \
+    Py_EXPORTED_SYMBOL uint32_t                                   \
+    required_hpy_major_version_##modname = HPY_ABI_VERSION_MAJOR; \
+    Py_EXPORTED_SYMBOL uint32_t                                   \
+    required_hpy_minor_version_##modname = HPY_ABI_VERSION_MINOR; \
```

Comment:
If I understood it correctly, those variables are read-only, right? In that case, I think they should be `const`.

Reply:
fixed

Reply:
Hm, `const` has a different semantics in C++ and gives compiler error in this context. We could use `#if defined(__cplusplus)`, but I was reading through the docs on what `const` actually means and implies and now it seems safer and easier to me to just not use it at all.